datafusion/common/src/scalar.rs

```diff
@@ -5067,76 +4680,45 @@ mod tests {
         assert_eq!(array, &expected);
     }
 
-    #[test]
-    fn test_nested_lists() {
-        // Define inner list scalars
-        let a1 = ListArray::from_iter_primitive::<Int32Type, _, _>(vec![Some(vec![
```

Comment:
This test is length > 1, which violate the ScalarValue::List constraint. So I rewrite the test.